### PR TITLE
Create hoursWorked API

### LINKED ISSUE #10: GET: `/api/hoursWorked?date=`
## API description
- URL query: (optional) id (A good reference on parameter query on [Stack Overflow]([https://stackoverflow.com/questions/150505/capturing-url-parameters-in-request-get)
- Authen header: Required.
- Body: None
- Description: This API will return the user's all `hoursWorked` in the form: 
```json
[
   {
      "date": "17/02/2023",
      "hoursWorked": 8.00,
   },
   {
      "date": "18/02/2023",
      "hoursWorked": 5.00,
    }
]
```
If `date` is specified, simply return an array with one record for that `date`.
Enhancement: Could you add two more queries `from` and `to` so that we can query every date in between? This can be magically done with Django ORM's queryset

### PR DESCRIPTION
**Pull request by mimiramita**
resolves #10 
I think my code should work. However, I don't know how to add a parameter to the API in Swagger, so I tested adding the parameter by directly call the API in the url bar. Please let me know if it worked.

### CODE REVIEW

File: account/views.py [hunk 1]
```diff
@@ -40,3 +40,16 @@ def post(self, request):
         except Exception as e:
             return Response({'status': False, 'message': str(e)},
                             status=status.HTTP_400_BAD_REQUEST)
+
```
Comment (nguyenvothuan) [inline account/views.py:49]: try looking for swagger_auto_schema. This should display the swagger ui. I would recommend ChatGPT for this

**Review by nguyenvothuan — CHANGES_REQUESTED** [review]

File: account/views.py [hunk 2]
```diff
@@ -3,40 +3,72 @@
...
+date = openapi.Parameter('date', in_=openapi.IN_QUERY,
+                           type=openapi.TYPE_STRING)
+
+
+class HoursWorked(APIView):
+    @swagger_auto_schema(
+        manual_parameters=[date],
```
Comment (nguyenvothuan) [inline account/views.py]: look great!

**Review by nguyenvothuan — COMMENTED** [review]
LGTM at the moment. Can you please create a video to demo how your API works?

**Comment by mimiramita** [comment]
I created a new time entry in my local database to test this out. The first part of the video is the case when the requested time entry exists. The second part is when it doesn't.

https://user-images.githubusercontent.com/111351821/233251002-b349b01c-6ea2-4dec-995e-2b580793e00b.mov

**Comment by nguyenvothuan** [comment]
> I created a new time entry in my local database to test this out. The first part of the video is the case when the requested time entry exists. The second part is when it doesn't.
> 
>  Screen.Recording.2023-04-19.at.11.25.23.PM.mov

Nice job Mimi. One thing left is that `date` query is optional. This means, if we don't insert the `date` query in the URL, the server still returns a 200 status response with an array of entries of all dates in the past. If this seems to be a lot, you may want to limit the number of entries to 10 or 20, whichever fits the bandwidth. Recheck the description in #10

File: account/views.py [hunk 3]
```diff
@@ -3,65 +3,127 @@
 from rest_framework import status
 from rest_framework.views import APIView
 from time_entry.models import TimeEntry
+from account.models import Account
 from account.utils import date_field_from_date
 from datetime import datetime
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema, APIView
```
Comment (nguyenvothuan) [inline account/views.py]: wait, why are we having 2 `APIView` here? On line 6 and line 10. This would definitely cause unexpected behaviors.

**Review by nguyenvothuan — APPROVED** [review]
lgtm

**Review by nguyenvothuan — APPROVED** [review]

**Pull request merged**